lib/Interpreter/DeviceKernelInliner.cpp

```diff
@@ -0,0 +1,30 @@
+//--------------------------------------------------------------------*- C++ -*-
+// CLING - the C++ LLVM-based InterpreterG :)
+// author:  Simeon Ehrig <s.ehrig@hzdr.de>
+//
+// This file is dual-licensed: you can choose to license it under the University
+// of Illinois Open Source License or the GNU Lesser General Public License. See
+// LICENSE.TXT for details.
+//------------------------------------------------------------------------------
+
+#include "DeviceKernelInliner.h"
+
+#include <clang/AST/Attr.h>
+
+#include <llvm/Support/Casting.h>
+
+namespace cling {
+
+DeviceKernelInliner::DeviceKernelInliner(clang::Sema *S) : ASTTransformer(S) {}
+
+ASTTransformer::Result DeviceKernelInliner::Transform(clang::Decl *D) {
+  clang::FunctionDecl *F = llvm::dyn_cast<clang::FunctionDecl>(D);
+  if (F && F->hasAttr<clang::CUDADeviceAttr>()) {
+    if (!F->isInlineSpecified()) {
```

Comment:
Should this be a performance/storage optimization or am I mistaken?

I have tested it and it works. So it's fine for me. I am also adding a second optimization. I have removed the check `!F->isInlineSpecified()` because the result is always the same. The variables `IsInlineSpecified` and `IsInline` of the function declaration are `true`. Setting a bool variable should be faster than an if statement.

Reply:
> Should this be a performance/storage optimization or am I mistaken?

No, this is about keeping variables at the closes scope possible, by moving the declaration of `F` into the `if` condition.